synapse/storage/databases/main/client_ips.py

```diff
@@ -555,8 +555,11 @@ async def get_last_client_ip_by_device(
         return ret
 
     async def get_user_ip_and_agents(
-        self, user: UserID
+        self, user: UserID, since_ts: Optional[float] = 0
```

Comment:
If it's an `Optional`, surely the default should be `None`? And why a `float` and not an `int`? `last_seen` is a `BIGINT`.

Reply:
Sorry, got confused when someone (I forgot whom) told me to "just use `time.time()`" for timestamps, which returns floating-point seconds instead of integer milliseconds.